pkg/models/event.go

```diff
@@ -27,37 +25,39 @@ import (
  * Event struct to hold event data
  */
 type Event struct {
-	ID       bson.ObjectId `bson:"_id,omitempty" json:"id"`
-	Pushed   int64         `bson:"pushed" json:"pushed"`
-	Device   string        `bson:"device" json:"device"` // Device identifier (name or id)
-	Created  int64         `bson:"created" json:"created"`
-	Modified int64         `bson:"modified" json:"modified"`
-	Origin   int64         `bson:"origin" json:"origin"`
-	Event    string        `bson:"event,omitempty" json:"event"` // Schedule event identifier
-	Readings []Reading     `bson:"readings" json:"readings"`     // List of readings
+	ID       string    `json:"id"`
+	Pushed   int64     `json:"pushed"`
+	Device   string    `json:"device"` // Device identifier (name or id)
+	Created  int64     `json:"created"`
+	Modified int64     `json:"modified"`
+	Origin   int64     `json:"origin"`
+	Event    string    `json:"event"`    // Schedule event identifier
+	Readings []Reading `json:"readings"` // List of readings
 }
 
 // Custom marshaling to make empty strings null
 func (e Event) MarshalJSON() ([]byte, error) {
 	test := struct {
-		ID       bson.ObjectId `json:"id"`
-		Pushed   int64         `json:"pushed,omitempty"`
-		Device   *string       `json:"device,omitempty"` // Device identifier (name or id)
-		Created  int64         `json:"created,omitempty"`
-		Modified int64         `json:"modified,omitempty"`
-		Origin   int64         `json:"origin,omitempty"`
-		Schedule *string       `json:"schedule,omitempty"` // Schedule identifier
-		Event    *string       `json:"event,omitempty"`    // Schedule event identifier
-		Readings []Reading     `json:"readings,omitempty"` // List of readings
+		ID       *string   `json:"id"`
```

Comment:
Can you add `omitempty` tag? If an event isn't stored in database, ID field always is empty. After this modifications, maybe you should update some tests.